### ISSUE TITLE
Bug: Packages cannot be imported into TypeScript projects if module is CommonJS

### ISSUE THREAD

**Issue by daniel7grant** [typescript-eslint/typescript-eslint#9767]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Issue Description

If you are developing a new typescript-eslint plugin, and using `module: CommonJS` in the `tsconfig.json` (default setting for the tsconfig), importing certain packages will fail (e. g. `@typescript-eslint/parser`, `@typescript-eslint/rule-tester` maybe more). The TypeScript error helpfully points towards the `moduleResolution` setting, but it might not be trivial to fix this problem (in my case I need CommonJS specifically for dual building CJS and ESM).

```sh
Cannot find module '@typescript-eslint/rule-tester' or its corresponding type declarations.
  There are types at '.../@typescript-eslint/rule-tester/dist/index.d.ts', but this result could not be resolved under your current 'moduleResolution' setting. Consider updating to 'node16', 'nodenext', or 'bundler'.
```

The source of the problem is a missing `main` field in the `package.json`. In the default TypeScript module resolver, `node10`, the `exports` field is ignored in compatibility with an older version of Node.js. In this case the import falls back to the `main` and `types` fields. The exports field should be duplicated into these fields for the widest compatibility.

You can also verify this problem by using the brilliant [Are The Types Wrong?](https://arethetypeswrong.github.io/?p=%40typescript-eslint%2Frule-tester%408.0.1) website, which shows a missing `node10` compatibility.

### Reproduction Repository Link

https://github.com/daniel7grant/eslint-install-debug

### Repro Steps

Create a new TypeScript project with default settings (`tsc --init`) and install and import `@typescript-eslint/parser`. The TypeScript building will throw an error noting that the module resolution failed.

```sh
mkdir eslint-installer
cd eslint-installer/
npm init -y
npm i typescript -D
npx tsc --init
npm i @typescript-eslint/parser 
echo 'import parser from "@typescript-eslint/parser";' > index.ts
npx tsc
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser`            | `8.0.1` |
| `@typescript-eslint/rule-tester`       | `8.0.1` |
| `TypeScript`                           | `5.5.4` |

**Comment by bradzacher** [comment]
> which shows a missing node10 compatibility

This is intentional.
As it's name suggests - `node10` emulates node v10's module resolution. We don't support node10 and it's long since EOL.

You need to use `node16` module resolution - which is fully compatible with commonjs and ESM.
